commit 03020ee16d982b8ccd106c153d6e243ba7d8ddab
Author: Sopheap15 <sopheap15@gmail.com>
Date:   Mon Oct 6 15:54:08 2025 +0700

    Improvement metadata and other base on anthony suggestion

diff --git a/pre_process_data/scripts/silva.R b/pre_process_data/scripts/silva.R
index 0b81abf..97473a0 100644
--- a/pre_process_data/scripts/silva.R
+++ b/pre_process_data/scripts/silva.R
@@ -1,5 +1,6 @@
 library(Seurat)
 library(dplyr)
+
 # set current working directory
 setwd("/srv/baia/prj/pvsca/pre_process_data")
 
@@ -7,13 +8,8 @@ df <- readxl::read_xls("data/TableS2_Cell_Metadata.xls", skip = 1)
 df <- df |>
   mutate(
     refine_state = case_when(
-      State == "Hypnozoite" ~ "Hypnozoite",
-      TRUE ~ topcell
-    ),
-    refine_state = case_when(
-      refine_state == "Male" ~ "Male gametocyte",
-      refine_state == "Female" ~ "Female gametocyte",
-      TRUE ~ refine_state
+      State == "Replicative" ~ "Szhizont (liver stage)",
+      TRUE ~ State
     )
   )
 
@@ -28,7 +24,7 @@ silva <- CreateSeuratObject(counts = raw)
 
 silva$study_pmid <- "39223117"
 silva$run_id <- paste0("SRR18134", 227:284)
-silva$study_label <- "Mancio_silva2022"
+silva$study_label <- "Mancio Silva2022"
 silva$num_srr <- 58
 silva$pub_year <- 2022
 silva$goegraphic_location <- "USA_RockvilleThailand_Ubon-Ratchathani"
@@ -39,9 +35,9 @@ silva$sequncer <- c(
 )
 silva$host_species <- "Homo sapiens"
 silva$host_id <- NA
-silva$sample_type <- "Infected host hepatocytes"
+silva$sample_type <- "Mammalian host: hepatocyte"
 silva$parasite_stage <- "Liver stage"
-silva$strain <- "patient isolate"
+silva$strain <- "Patient isolate"
 
 silva$day_post_infection <- c(
   rep(1, 2),
