commit 5e8e1441d9235db3dd37d678f48de053f9b8dfd6
Author: Sopheap15 <sopheap15@gmail.com>
Date:   Thu Oct 9 08:36:04 2025 +0700

    fix dropdown box to show genes name in embedding

diff --git a/pre_process_data/scripts/silva.R b/pre_process_data/scripts/silva.R
index 42b2877..4356988 100644
--- a/pre_process_data/scripts/silva.R
+++ b/pre_process_data/scripts/silva.R
@@ -73,6 +73,8 @@ for (cell in df$Updated_names) {
   silva$refine_state[matched] <- df$refine_state[df$Updated_names == cell]
 }
 
+# Add barcode column for consistency with other studies
+silva$barcode <- colnames(silva)
 
 # match_cell <- intersect(colnames(silva), df$Updated_names)
 # silva$refine_state[match_cell] <- df$refine_state[match(
